src/datasets/search.py

```diff
@@ -447,6 +447,7 @@ def add_faiss_index(
         string_factory: Optional[str] = None,
         metric_type: Optional[int] = None,
         custom_index: Optional["faiss.Index"] = None,
+        batch_size: Optional[int] = 1000,
```

Comment:
I think you need to make the same changes in arrow_dataset.py, where add_faiss_index and add_faiss_index_from_external_arrays are overriden ;)